play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/govukfrontend/views/components/GovukServiceNavigationSpec.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.govukfrontend.views
+package components
+
+import play.api.i18n.{Lang, Messages}
+import uk.gov.hmrc.helpers.MessagesSupport
+import play.api.mvc.RequestHeader
+import play.api.test.FakeRequest
+import play.twirl.api.HtmlFormat
+import uk.gov.hmrc.govukfrontend.views.html.components._
+
+import scala.util.Try
+
+class GovukServiceNavigationSpec
+    extends TemplateUnitBaseSpec[ServiceNavigation]("govukServiceNavigation")
+    with MessagesSupport {
+  private val component = app.injector.instanceOf[GovukServiceNavigation]
+
+  def render(templateParams: ServiceNavigation): Try[HtmlFormat.Appendable] = Try(component(templateParams))
+
+  "serviceNavigation" should {
+    "render custom service name correctly" in {
+      val params = ServiceNavigation(serviceName = Some("my-service"))
+      val output = component(params).select(".govuk-service-navigation__text")
+
+      output.first().text() shouldBe "my-service"
+    }
+
+    "render ServiceNavigationItems correctly" in {
+      val params = ServiceNavigation(
+        navigation = Seq(
+          ServiceNavigationItem(
+            content = Text("Cupcakes")
+          )
+        )
+      )
+      val output = component(params).select(".govuk-service-navigation__text")
+
+      output.first().text() shouldBe "Cupcakes"
+    }
+
+    "render ServiceNavigationItems with links correctly" in {
+      val params = ServiceNavigation(
+        serviceName = Some("cupcakes-service"),
+        navigation = Seq(
+          ServiceNavigationItem(
+            content = Text("Cupcakes"),
+            href = "#"
+          )
+        )
+      )
+      val output = component(params).select(".govuk-service-navigation__link")
+
+      output.first().text()       shouldBe "Cupcakes"
+      output.first().attr("href") shouldBe "#"
+    }
+
+    "render ServiceNavigation with slotted content" in {
+      val params = ServiceNavigation(
+        serviceName = Some("cupcakes-service"),
+        slots = Some(
+          ServiceNavigationSlot(
+            start = Some("<div class=\"my-custom-class\">Cupcakes are delicious!</div>")
+          )
+        )
+      )
+      val output = component(params).select(".my-custom-class")
+
+      output.first().text() shouldBe "Cupcakes are delicious!"
+    }
```

Comment:
This was done because we needed a unit test to be included as we are now using `with MessagesSupport` in our unit test. As such, wrote at least one unit test per model. This also allowed me to spot that this component nearly didn't have an `Alias` for the `ServiceNavigationSlots`, which would have meant end users would have needed to import that model 😅